webkit/main.c

```diff
@@ -10,141 +10,162 @@
+
+  struct sigaction intHandler;
+  intHandler.sa_handler = int_handler;
+  sigemptyset(&intHandler.sa_mask);
+  intHandler.sa_flags = 0;
+  sigaction(SIGINT, &intHandler, NULL);
+
+  // Create the window, set basic properties
+  window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+  gtk_window_set_position(GTK_WINDOW(window), GTK_WIN_POS_NONE);
+  gtk_window_set_default_size(GTK_WINDOW(window), width, height);
+  gtk_window_move(GTK_WINDOW(window), x, y);
+  gtk_window_set_title(GTK_WINDOW(window), title);
+  gtk_widget_add_events(GTK_WIDGET(window), GDK_CONFIGURE);
+  g_signal_connect(G_OBJECT(window), "delete-event",
+                   G_CALLBACK(handle_gtk_quit), NULL);
+  gtk_widget_set_app_paintable(window, TRUE);
+
+  // Set up a callback to react to screen changes
+  g_signal_connect(G_OBJECT(window), "screen-changed",
+                   G_CALLBACK(screen_changed), NULL);
+
+  // Set up a callback to handle resize/move events
+  g_signal_connect(G_OBJECT(window), "configure-event",
+                   G_CALLBACK(configure_callback), NULL);
+
+  // Set up and add the WebKit web view widget
+  // Disable caching
+  WebKitWebContext *wk_context = webkit_web_context_get_default();
+  webkit_web_context_set_cache_model(wk_context,
+                                     WEBKIT_CACHE_MODEL_DOCUMENT_VIEWER);
+
+  WebKitSecurityManager *security_manager =
+      webkit_web_context_get_security_manager(wk_context);
+  webkit_security_manager_register_uri_scheme_as_secure(security_manager, "ws");
+
+  WebKitWebView *web_view =
+      WEBKIT_WEB_VIEW(webkit_web_view_new_with_context(wk_context));
+
+  WebKitSettings *wk_settings = webkit_settings_new();
+  webkit_settings_set_enable_write_console_messages_to_stdout(wk_settings,
+                                                              true);
+  printf("enable webaudio\n");
+  webkit_settings_set_enable_webaudio(wk_settings, true);
+  webkit_settings_set_enable_webgl(wk_settings, true);
+  webkit_settings_set_media_playback_requires_user_gesture(wk_settings, false);
+  webkit_settings_set_media_playback_allows_inline(wk_settings, true);
+  webkit_settings_set_enable_media_stream(wk_settings, true);
+  webkit_settings_set_enable_mediasource(wk_settings, true);
+  webkit_settings_set_enable_media_capabilities(wk_settings, true);
+  webkit_settings_set_allow_file_access_from_file_urls(wk_settings, true);
+  webkit_settings_set_allow_universal_access_from_file_urls(wk_settings, true);
+  webkit_settings_set_enable_media(wk_settings, true);
+  webkit_web_view_set_settings(web_view, wk_settings);
+  // Make transparent
+  GdkRGBA rgba = {.alpha = 0.0};
+  webkit_web_view_set_background_color(web_view, &rgba);
+  gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(web_view));
+
+  // Load the specified URI
+  webkit_web_view_run_javascript(
+      web_view,
+      "window.AudioContext = class extends window.webkitAudioContext { "
+      "constructor() { super(); this.resume(); return this; } }",
```

Comment:
Nitpicking here, but the `return this;` in that javascript is completely pointless.

Reply:
well, it didnt work before putting this in for me